tests/multipleDeclarationsWithDifferentTypes.ts

```diff
@@ -0,0 +1,16 @@
+var s = 'test';
+
+var s1: string = 'test';
+
+var s2 = 'test';
+var s2: string = 'test';
+
+var s3: number = 2;
+var s3 = 'test';
+
+var s4: string = 'test';
+var s4: number = 2;
```

Comment:
interesting choice not to have an error on `2`. Probably the right one, to avoid inundating the user with errors.

Reply:
Do you mean this?

```ts
var s4: string = 'test';
var s4: number = 2; // this has an error because it should actually be a string
var s4: string = 's'; // it doesn't produce an error because it compares only with the first var declaration
```

If so, I got this behavior from TS itself 😅 

> to avoid inundating the user with errors.

^ That makes sense to me!

Reply:
I meant no error on the `2` in `var s4: number = 2`, even though `s4` actually has type string. It makes sense because of the explicit type annotation `: number`.